--- a/Graph/Easy/As_graphml.pm	2008-06-07 09:47:31.000000000 +0100
+++ b/Graph/Easy/As_graphml.pm	2010-03-21 01:07:56.000000000 +0000
@@ -313,10 +313,10 @@
   my ($self, $indent, $ids) = @_;
 
   my $g = $self->{graph};
-  my $txt = $indent . '<node id="' . $g->_safe_xml($self->{name}) . "\"/>\n";
+  my $txt = $indent . '<node id="' . $g->_safe_xml($self->{name}) . "\">\n";
 
   $txt .= $g->_attributes_as_graphml($self, $indent, $ids->{node});
-
+  $txt .= $indent . "</node>\n";
   $txt;
   }
 
@@ -332,10 +332,10 @@
 
   my $g = $self->{graph};
   my $txt = $indent . '<edge source="' . $g->_safe_xml($self->{from}->{name}) . 
-		     '" target="' . $g->_safe_xml($self->{to}->{name}) . "\"/>\n";
+		     '" target="' . $g->_safe_xml($self->{to}->{name}) . "\">\n";
 
   $txt .= $g->_attributes_as_graphml($self, $indent, $ids->{edge});
-
+  $txt .= $indent . "</edge>\n";
   $txt;
   }
  
